src/server/main_service.cc

```diff
@@ -1167,12 +1167,19 @@ void Service::Exec(CmdArgList args, ConnectionContext* cntx) {
 
 void Service::Publish(CmdArgList args, ConnectionContext* cntx) {
   string_view channel = ArgS(args, 1);
-  string_view message = ArgS(args, 2);
+
+  // To ensure that this message that we are process in one context will live long enough
+  // when passing over to other contexts we need to have shared ownership.
+  // This use to have a manual ref counter. But it is replaced here with shared pointer that
```

Comment:
I do not think that history matters here because the readers won't understand what you are referring to.
Maybe replace line 1173-1176 with 
```
shared_ptr ensures that the message lives until it's been sent to all subscribers and handled by DispatchOperations. 
```

Reply:
ok